### PR TITLE
chore(edc-client): Remove wildcard policy check

### PR DESCRIPTION
**Pull request by ds-jhartmann**

### CODE REVIEW

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerService.java
Comment (dsmf) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerService.java:1]: Hi @ds-jhartmann, What was the requirement / story? 

Judged by the heading of the PR the code change looks good to me, but I don't understand the professional background without more details / link to requirement. 

And shouldn't the change be added to the changelog?

File: irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerServiceTest.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ void shouldConfirmValidPolicyWhenWildcardIsSet() {
         boolean result = policyCheckerService.isValid(policy);
 
         // then
-        assertThat(result).isTrue();
+        assertThat(result).isFalse();
```
Comment (dsmf) [inline irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerServiceTest.java:101]: Hi @ds-jhartmann, The assertion was negated here, but the test name is still the old one and does not reflect the new behaviour

**Review by dsmf — CHANGES_REQUESTED** [review]
see comments

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerService.java
Comment (ds-jhartmann) [reply irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerService.java:1]: Adding this in the first place should have been a temporary workaround in the first version of the policy store. This should have been removed ages ago :D
It was never a feature, so mentioning this in the changelog is not necessary IMO

File: irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerServiceTest.java (hunk 1 above)
Comment (ds-jhartmann) [reply irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerServiceTest.java:101]: Good point but I think I will remove the test entirely

Comment (ds-jhartmann) [reply irs-edc-client/src/test/java/org/eclipse/tractusx/irs/edc/client/policy/PolicyCheckerServiceTest.java:101]: Removed the test entirely

**Pull request merged**